apiary-receiver/apiary-receiver-sqs/src/main/java/com/expedia/apiary/extensions/receiver/sqs/messaging/SqsMessageReader.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (C) 2018-2019 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.apiary.extensions.receiver.sqs.messaging;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Iterator;
+import java.util.Optional;
+
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.AmazonSQSClientBuilder;
+import com.amazonaws.services.sqs.model.DeleteMessageRequest;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.expedia.apiary.extensions.receiver.common.messaging.JsonMetaStoreEventDeserializer;
+import com.expedia.apiary.extensions.receiver.common.messaging.MessageDeserializer;
+import com.expedia.apiary.extensions.receiver.common.messaging.MessageReader;
+import com.expedia.apiary.extensions.receiver.common.error.SerDeException;
+import com.expedia.apiary.extensions.receiver.common.event.ListenerEvent;
+
+public class SqsMessageReader implements MessageReader {
+  private static final Integer DEFAULT_POLLING_WAIT_TIME = 10;
+  private static final Integer DEFAULT_MAX_MESSAGES = 10;
+
+  private String queueUrl;
+  private int waitTimeSeconds;
+  private int maxMessages;
+  private MessageDeserializer messageDeserializer;
+  private AmazonSQS consumer;
+  private Iterator<Message> records;
+
+  private SqsMessageReader(String queueUrl, int waitTimeSeconds, int maxMessages,
+                           MessageDeserializer messageDeserializer, AmazonSQS consumer) {
+    this.queueUrl = queueUrl;
+    this.waitTimeSeconds = waitTimeSeconds;
+    this.maxMessages = maxMessages;
+    this.messageDeserializer = messageDeserializer;
+    this.consumer = consumer;
+  }
+
+  @Override
+  public void close() {
+    consumer.shutdown();
+  }
+
+  @Override
+  public Optional<ListenerEvent> read() {
+    if (records == null || !records.hasNext()) {
+      records = receiveMessage();
+    }
+    if (records.hasNext()) {
+      return Optional.of(eventPayLoad(records.next()));
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  public void delete(Message message) {
```

Comment:
For now we should delete on read in order to keep existing functionality, future work should look at a better way of allowing us to only delete when the downstream usage of the message has completed successfully.

Reply:
Added the delete on read back in, thanks!